draft-ietf-tls-esni.md

```diff
@@ -430,6 +430,35 @@ the ClientHelloOuter. The server then computes the digest of the reconstructed
 ClientHelloInner. If the digest does not equal OuterExtensions.inner_digest,
 then the server MUST abort the connection with a "decrypt_error" alert.
 
+# The "handshake_padding" Extension {#handshake-padding}
+
+Encrypting the ClientHelloInner may not be sufficient to protect sensitive
+parameters. In particular, the length of the ClientHelloInner, or the length of
+the backend server's response, may leak information about these parameters. The
+client and backend server can use the "handshake_padding" extension in order to
+mitigate this leakage.
+
+A new extension type is defined, which the client MAY include in its
+ClientHello.
```

Comment:
Should this be ClientHellIoInner?

Reply:
IIRC the intent is for the the "handshake_padding" to be general purpose, i.e., not specific to ECH.